data/Posts/express-router.js

```diff
@@ -0,0 +1,132 @@
+const express = require('express');
+const Data = require('../db');
+
+const router = express.Router();
+
+//GET
+
+router.get('/', async (req, res) => {
+  try {
+    const posts = await Data.find(req.query);
+    res.status(200).json(posts);
+  } catch (error) {
+    // log error to database
+    console.log(error);
+    res.status(500).json({
+      message: 'Error retrieving the data'
+    });
+  }
+});
+
+router.get('/:id', async (req, res) => {
+  try {
+    const post = await Data.findById(req.params.id);
+    if (post) {
+      res.status(200).json(post);
+    } else {
+      res.status(404).json({
+        success: false,
+        message: 'The post with the specified ID does not exist.'
+      });
+    }
+  } catch (error) {
+    res.status(500).json({
+      success: false,
+      error: 'The information could not be retrieved.'
+    });
+  }
+});
+
+router.get('/:id/comments', async (req, res) => {
+  try {
+    const comments = await Data.findPostComments(req.body);
+    if (comments) {
+      res.status(200).json(comments);
+    } else {
+      res.status(404).json({
+        success: false,
+        message: 'The post with the specified ID does not exist.'
```

Comment:
might want to validate the Id to be sure it exists with middleware or by a find request.

i.e. 

```
const found = await Data.findById(id);
if(!found) {
   res.....404
} else {
    await Data.findPostComments....your logic here.
}

```

when you get to the error handling here, you know the id exists....what you are handling is whether or not you have comments on the page. So your handler should say something to the effect that there are no comments associated with that user id.